signer/src/stacks/contracts.rs

```diff
@@ -339,21 +339,19 @@ impl CompleteDepositV1 {
             return Err(DepositErrorMsg::DeployerMismatch.into_error(req_ctx, self));
         }
         // 2. Check that the signer has a record of the deposit request
-        //    from our list of pending and accepted deposit requests.
-        //
-        // Check that this is actually a pending and accepted deposit
-        // request.
+        //    from our list of swept deposit requests.
         let deposit_requests = db
-            .get_pending_accepted_deposit_requests(
-                &req_ctx.chain_tip.block_hash,
-                req_ctx.context_window,
-                req_ctx.signatures_required,
-            )
+            .get_swept_deposit_requests(&req_ctx.chain_tip.block_hash, req_ctx.context_window)
             .await?;
 
         let deposit_request = deposit_requests
             .into_iter()
-            .find(|req| req.outpoint() == self.outpoint)
+            .find(|req| req.deposit_outpoint() == self.outpoint)
+            .ok_or_else(|| DepositErrorMsg::RequestMissing.into_error(req_ctx, self))?;
+
+        let deposit_request = db
+            .get_deposit_request(&deposit_request.txid, deposit_request.output_index)
```

Comment:
Having a `get_deposit_request` seems pretty useful, but in this case I think it would be better to add a `max_fee` field to the `SweptDepositRequest` object. This way we don't need to do the extra query at all.

Reply:
Makes sense: since `SweptDepositRequest` was missing some info wrt `DepositRequest` I initially thought we may need more of them during validation; but it seems the missing fields are more "sweeping" related, so at this point we don't really need them. Removed in favor of adding `max_fee` in [9af3a56](https://github.com/stacks-network/sbtc/pull/764/commits/9af3a56a1b05d9ca7208a3660db447eff825b74d).